src/software/time/time.cpp

```diff
@@ -23,3 +20,27 @@ double Time::getMilliseconds() const
 }
 
 Time::~Time() {}
+
+double Time::now()
+{
+    const auto clock_now = std::chrono::system_clock::now();
+    return static_cast<double>(std::chrono::duration_cast<std::chrono::microseconds>(
+                                   clock_now.time_since_epoch())
+                                   .count()) /
+           1000000.0;
+}
+
+double Time::secondsSince(std::chrono::time_point<std::chrono::system_clock> start_time)
+{
+    return millisecondsSince(start_time) / 1000.0;
+}
+
+double Time::millisecondsSince(
+    std::chrono::time_point<std::chrono::system_clock> start_time)
```

Comment:
If we end up going with my above suggestion, we should be able to delete this function.